packages/stargate/src/signingstargateclient.spec.ts

```diff
@@ -321,6 +322,86 @@ describe("SigningStargateClient", () => {
     });
   });
 
+  describe("sendIbcTokens", () => {
+    it("works with direct signing", async () => {
+      pendingWithoutSimapp();
+      const wallet = await DirectSecp256k1HdWallet.fromMnemonic(faucet.mnemonic);
+      const client = await SigningStargateClient.connectWithSigner(simapp.tendermintUrl, wallet);
+      const memo = "Cross-chain fun";
+
+      // both timeouts set
+      {
+        const result = await client.sendIbcTokens(
+          faucet.address0,
+          faucet.address1,
+          coin(1234, "ucosm"),
+          "fooPort",
+          "fooChannel",
+          { revisionHeight: Long.fromNumber(123), revisionNumber: Long.fromNumber(456) },
+          Math.floor(Date.now() / 1000) + 60,
+          memo,
+        );
+        // CheckTx must pass but the execution must fail in DeliverTx due to invalid channel/port
```

Comment:
Would you get another response if CheckTx fails? Eg. would `sendIbcTokens` throw on CheckTx error?

Is there any way you can ensure this is due to DeliverTx not CheckTx?

Reply:
Yes, `await client.sendIbcTokens` throws and error on submission fails (CheckTx) but not on execution fails (DeliverTx).